src/libsystemd/libsystemd.sym

```diff
@@ -998,6 +998,7 @@ global:
         sd_varlink_invoke;
         sd_varlink_invokeb;
         sd_varlink_is_idle;
+        sd_varlink_is_processing_method;
```

Comment:
I don't follow this. Can you please clarify

Reply:
As explained in a later commit, if we send an error code, then we should not send any message even with _MORE flag is set.

Reply:
Sorry, but I still don't understand.

Should I use the same approach in other methods?

Reply:
`list_machine_one_and_maybe_read_metadata()` returns non-negative value when
1. an error occurred and `sd_varlink_error()` is called,
2. a (non-error) message is correctly sent with `sd_varlink_notify()` or `sd_varlink_reply()`.

On case 1, we cannot send any messages after that, so we need to skip all subsequent machines.
On case 2, we can send more messages, of course.

So, we need to distinguish the current state of the varlink.

`list_image_one_and_maybe_read_metadata()` does not call `sd_varlink_error()`, so the change is not necessary, but for safety.

Reply:
Thanks for the explanation

Reply:
btw, why not simply checking `r != 0` in relevant loop when listing machines and images?

https://github.com/systemd/systemd/pull/34882/files#diff-15c4a737d22304facac2b3af8fc398e63e92c1e3ed63b4bc975a853cf8c7ab17L700-L702